oso_sdk/types/conversion.py

```diff
@@ -0,0 +1,74 @@
+from typing import Optional, Tuple, Any
+from oso_cloud import Fact, Value as OsoValue
+
+
+def _process_class(cls, name=None, id=None):
```

Comment:
I'd like to understand possible failure modes of this a bit better (probably a job for tests)

Reply:
though fwiw this is pretty much a really basic ripoff of the `dataclass` implementation